numpy/core/tests/test_multiarray.py

```diff
@@ -4818,7 +4818,7 @@ def setup(self):
         self.x[0,:, 1] = [np.nan, np.inf, -np.inf, np.nan]
         self.dtype = self.x.dtype
         self.tempdir = tempfile.mkdtemp()
-        self.filename = tempfile.mktemp(dir=self.tempdir)
+        self.filename = tempfile.mkstemp(dir=self.tempdir)[1]
```

Comment:
As [documented](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp)
> `mkstemp()` returns a tuple containing an OS-level handle to an open file (as would be returned by `os.open()`) and the absolute pathname of that file, in that order.

So someone needs to close that open handle. I would prefer to decouple all the tests that use `self.filename` by converting them to tests that use pytest's `tmp_file` instead. Then each test would get a new filename: one failing test would not influence another.